packages/rum-angular/src/apm.service.ts

```diff
@@ -24,19 +24,19 @@
  */
 
 import { Router, NavigationStart } from '@angular/router'
-import { Injectable } from '@angular/core'
+import { Inject, Injectable } from '@angular/core'
 import { afterFrame } from '@elastic/apm-rum-core'
+import { ApmBase } from '@elastic/apm-rum'
+import { APM } from './apm.module'
 
 @Injectable({
   providedIn: 'root'
 })
 export class ApmService {
-  static apm: any
-
-  constructor(public router: Router) {}
+  constructor(@Inject(APM) public apm: ApmBase, public router: Router) {}
```

Comment:
I marked router private in mine since it's not needed from clients calling this service. `private readonly router: Router`

Reply:
Good point. It doesnt need to be public.